librz/core/canalysis.c

```diff
@@ -5560,29 +5560,39 @@ static void sdb_concat_by_path(Sdb *s, const char *path) {
 	sdb_free(db);
 }
 
-RZ_API void rz_core_analysis_cc_init(RzCore *core) {
+RZ_API bool rz_analysis_cc_init(RzCore *core, RZ_NULLABLE const char *path) {
 	const char *analysis_arch = rz_config_get(core->config, "analysis.arch");
 	Sdb *cc = core->analysis->sdb_cc;
-	if (!strcmp(analysis_arch, "null")) {
-		sdb_reset(cc);
-		RZ_FREE(cc->path);
-		return;
-	}
-
 	int bits = core->analysis->bits;
-	char *types_dir = rz_path_system(RZ_SDB_TYPES);
-	char *home_types_dir = rz_path_home_prefix(RZ_SDB_TYPES);
+	char *types_dir;
+	char *home_types_dir;
+	char *dbpath;
+	char *dbhomepath;
 	char buf[40];
-	char *dbpath = rz_file_path_join(types_dir, rz_strf(buf, "cc-%s-%d.sdb", analysis_arch, bits));
-	char *dbhomepath = rz_file_path_join(home_types_dir, rz_strf(buf, "cc-%s-%d.sdb", analysis_arch, bits));
-	free(types_dir);
-	free(home_types_dir);
+
+	if (path[0] == '/') {
+		types_dir = path;
+		home_types_dir = "";
+	} else {
+		types_dir = rz_path_system(path);
+		home_types_dir = rz_path_home_prefix(path);
+	}
```

Comment:
Can I suggest we just provide `path` and `homepath` as argument of this function instead of doing this thing? It seems a bit hacky and I'm not sure it works in all configurations (e.g. when building Rizin in portable mode).

Reply:
agreed